pkg/providers/amifamily/resolver.go

```
@@ -119,6 +119,9 @@ func (r Resolver) Resolve(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTem
 	if err != nil {
 		return nil, err
 	}
+	if len(amis) == 0 {
+		logging.FromContext(ctx).Errorf("no amis exist given constraints")
```

To comply with existing behavior, this should return an error.